### PR TITLE
feat: eslint integrated [#24]

### PR DESCRIPTION
**Pull request by anjaliahlawat**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+            "error",
+            "never"
+        ],
+        "template-tag-spacing": "error",
+        "unicode-bom": [
+            "error",
+            "never"
+        ],
+        "vars-on-top": "off",
+        "wrap-iife": "off",
```
Comment (h2grover) [inline .eslintrc.json]: @anjaliahlawat please airbnb eslint rule guide. It doesn't make sense to write so many custom rules

**Review by h2grover — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 1 above)
Comment (anjaliahlawat) [reply .eslintrc.json]: @h2grover I am going through the rules of eslint. But to be specific, which rules should I add ?

Comment (h2grover) [reply .eslintrc.json]: @anjaliahlawat what do you mean by that?
Please use default rules provided by airbnb plugin for now.

File: package.json [hunk 2]
```diff
@@ -19,6 +20,10 @@
     "@commitlint/cli": "^12.0.1",
     "@commitlint/config-conventional": "^12.0.1",
     "eslint": "^7.22.0",
+    "eslint-config-airbnb": "^18.2.1",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
```
Comment (h2grover) [inline package.json]: Why this?

File: package.json [hunk 3]
```diff
@@ -19,6 +20,10 @@
...
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-react": "^7.23.1",
```
Comment (h2grover) [inline package.json]: Why this?

File: src/tests/amazon/step_definition/login_implementation.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ var pageWrapper = {}
 // Gauge step implementations
 // --------------------------
 
-beforeSpec(async () => {
+beforeSpec(async function(){
```
Comment (h2grover) [inline src/tests/amazon/step_definition/login_implementation.js]: Why not arrow function here?

File: .eslintrc.json (hunk 1 above)
Comment (anjaliahlawat) [reply .eslintrc.json]: @h2grover I just thought of adding custom rules like camel casing etc

File: yarn-error.log [hunk 5]
```diff
@@ -1,8 +1,8 @@
 Arguments:
```
Comment (h2grover) [inline yarn-error.log]: Add this to gitignore and remove the file

File: src/tests/globalHooks.js [hunk 6]
```diff
@@ -2,19 +2,19 @@ var Setup = require("./setup")
...
 var page = {}
 
-beforeSuite(async function(){
-    setup = new Setup();
-    page = await setup.createPage();
-    gauge.dataStore.specStore.put("page", page)
-})
+beforeSuite(async () => {
+        setup = new Setup();
+        page = await setup.createPage();
```
Comment (h2grover) [inline src/tests/globalHooks.js]: Is it because of linting done by eslint?

File: package.json [hunk 7]
```diff
@@ -19,6 +20,10 @@
     "@commitlint/cli": "^12.0.1",
     "@commitlint/config-conventional": "^12.0.1",
     "eslint": "^7.22.0",
+    "eslint-config-airbnb": "^18.2.1",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-import": "^2.22.1",
```
Comment (h2grover) [inline package.json:21]: What is the usage?

File: package.json [hunk 8]
```diff
@@ -2,7 +2,8 @@
   "name": "gauge-js-template",
   "description": "Starter template for writing JavaScript tests for Gauge",
   "scripts": {
-    "test": "./node_modules/.bin/gauge run -p specs/"
+    "test": "./node_modules/.bin/gauge run -p specs/",
+    "lint": "eslint src/"
```
Comment (h2grover) [inline package.json]: Use node modules eslint and run it on root folder.

**Review by h2grover — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (anjaliahlawat) [reply package.json]: As I tried to run eslint command from terminal, it threw this error. I googled it but then it only worked when I added these dependencies.

File: package.json (hunk 2 above)
Comment (anjaliahlawat) [reply package.json]: @h2grover I was supposed to config airbnb with eslint. So I found this solution online.

File: src/tests/amazon/step_definition/login_implementation.js (hunk 4 above)
Comment (anjaliahlawat) [reply src/tests/amazon/step_definition/login_implementation.js]: @h2grover I guess I changed it when trying to resolve errors when I ran eslint. No problem I can switch it back to arrow functions. I was trying to understand how eslint is responding to different version of ECMA.

File: src/tests/globalHooks.js (hunk 6 above)
Comment (anjaliahlawat) [reply src/tests/globalHooks.js]: Yeah maybe. Also, I have added a support for es5 and es6 currently. Should I only use one of them?

**Comment by h2grover** [comment]
@anjaliahlawat I have added commit for `eslint` config and with `airbnb`. I have also added `prettier` plugin which I was going to ask you next but don't worry on that now.

Now you have to just run `yarn run lint` command on local and resolve those errors. Which are something like this

image

File: yarn-error.log (hunk 5 above)
Comment (h2grover) [reply yarn-error.log]: Also, push commit to remove this file

File: package.json (hunk 2 above)
Comment (h2grover) [reply package.json]: @anjaliahlawat 
not every solution is recommended and best of our interest. Please think twice before pushing any random code. And be prepared for the answers. You cannot say "**I found this solution online**"

Comment (anjaliahlawat) [reply package.json]: Yes, thanks. I'll keep that in mind.

**Comment by h2grover** [comment]
@anjaliahlawat Please donot work in parallel, work on this ticket first

File: src/lib/PageWrapper.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  async clickAndWaitForNavigation(id) {
+    await Promise.all([
+      this.page.waitForNavigation(),
+      await this.page.click(id),
+    ]);
+  }
+
+  getUrl() {
```
Comment (h2grover) [inline src/lib/PageWrapper.js:15]: @anjaliahlawat  this should be async function

File: yarn-error.log (hunk 5 above)
Comment (h2grover) [reply yarn-error.log]: @anjaliahlawat why this file is still here after mentioning twice?

**Review by h2grover — CHANGES_REQUESTED** [review]

**Comment by h2grover** [comment]
@anjaliahlawat another tip, when I request changes then please push all the commits at once not one by one.

**Comment by h2grover** [comment]
Let me know if this is https://github.com/anjaliahlawat/amazon-login-script/pull/37#issuecomment-806476947 not clear

File: src/lib/PageWrapper.js (hunk 9 above)
Comment (anjaliahlawat) [reply src/lib/PageWrapper.js:15]: I don't think so. Actually eslint showed redundant await error here. So I removed async-await from here and ran tests to check. The tests worked fine. So I think we don't need to make it async fn.

File: yarn-error.log (hunk 5 above)
Comment (anjaliahlawat) [reply yarn-error.log]: My apologies. I forgot to remove it from the staging area.

**Comment by anjaliahlawat** [comment]
All changes are done from my side. Please review them.

**Pull request merged**